content/languages/cpp/authoring/index.md

```diff
@@ -218,142 +261,118 @@ C++ sometimes requires some boilerplate code to implement non-trivial tests, che
     
-    //kata requires the input to be mutated, so tests need to copy it, because
-    //input array is used after calling user and reference solution
-    memcpy(reference, input, sizeof(double) * array_size);
-    square_every_item_ref(reference, array_size);
-    
-    //copy is made from original input, and not from an array fed to
-    //the reference solution
-    memcpy(actual, input, sizeof(double) * array_size);    
-    square_every_item(actual, array_size);
-
-    //assertion uses custom message to avoid confusing test output
-    //it also uses data from original, non-mutated input array
-    size_t invalid_position = get_mismatch_position(actual, reference, array_size);
-    cr_assert_arr_eq_cmp(actual, reference, array_size, cmp_double_fuzzy_equal,
-                         "Invalid answer at position %zu for input value %f, expected %f but got %f",
-                         invalid_position, 
-                         input[invalid_position], 
-                         reference[invalid_position], 
-                         actual[invalid_position]);
+    std::vector<int> result = numbers;
+    std::transform(result.begin(), result.end(), result.begin(), [](int a) { return a * a; });
+    return result;
   }
-}
-
-//a set of large random tests, with not so detailed debugging messages
-Test(random_tests, large_arrays) {
   
-  double array[1000];     //small enough to be allocated on the stack,
-  double reference[1000]; //but you can use dynamic memory if necessary.
+  std::string stringify_input(const std::vector<int>& input) {
+    return fmt::format("{}", fmt::join(input, ","));
+  }
   
-  for(int i=0; i<10; ++i) {
-    
-    //generate test cases
-    size_t array_size = rand() % 200 + 801;
-    fill_random_array(array, array_size);
-    
-    //since original array is not used after tests, it's enough to create only one copy
-    memcpy(reference, array, sizeof(double) * array_size);
+public:  
+  
+  //a set of small random tests, with verbose debugging messages
+  It(SmallArrays) {
     
-    square_every_item_ref(reference, array_size);
-    square_every_item(array, array_size);
+    for(int i=0; i<10; ++i) {
+      
+      //generate test case
+      size_t input_size = gen_small_size();
+      auto input = generate_random_input(input_size);
+      
+      //tests need to copy the input vector, because
+      //it is used after calling user and reference solution
+      auto original = input;
+      auto expected = square_every_item_ref(input);      
+      
+      auto actual = square_every_item(input);
+  
+      //assertion uses custom message to avoid confusing test output
+      //it also uses data from original, non-mutated input array
+      Assert::That(actual, EqualsContainer(expected), ExtraMessage(fmt::format("Input: {}", stringify_input(original))));
+    }
+  }  
+
+  //a set of large random tests, with not so detailed debugging messages
```

Comment:
```suggestion
      // tests need to copy the input vector, because
      // it is used after calling the user and reference solution
      auto original = input;
      auto expected = square_every_item_ref(input);      
      
      auto actual = square_every_item(input);
  
      // assertion uses a custom message to avoid confusing test output.
      // it also uses data from original, non-mutated input array
      Assert::That(actual, EqualsContainer(expected), ExtraMessage(fmt::format("Input: {}", stringify_input(original))));
    }
  }  

  // a set of large random tests, with less detailed debugging messages
```